tests/geometry/test_intersections.py

```diff
@@ -40,11 +41,286 @@ def test_one_and_two():
+    assert pairs.size == 0
+
+
+def test_intersection_two_lines():
+    x_min = np.array([0, 1])
+    x_max = np.array([2, 3])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 2
+    assert pairs[0, 0] == 0
+    assert pairs[1, 0] == 1
+
+
+def test_intersection_two_lines_switched_order():
+    x_min = np.array([1, 0])
+    x_max = np.array([3, 2])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 2
+    assert pairs[0, 0] == 0
+    assert pairs[1, 0] == 1
+
+
+def test_intersection_two_lines_same_start():
+    x_min = np.array([0, 0])
+    x_max = np.array([3, 2])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 2
+    assert pairs[0, 0] == 0
+    assert pairs[1, 0] == 1
+
+
+def test_intersection_two_lines_same_end():
+    x_min = np.array([0, 1])
+    x_max = np.array([3, 3])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 2
+    assert pairs[0, 0] == 0
+    assert pairs[1, 0] == 1
+
+
+def test_intersection_two_lines_same_start_and_end():
+    x_min = np.array([0, 0])
+    x_max = np.array([3, 3])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 2
+    assert pairs[0, 0] == 0
+    assert pairs[1, 0] == 1
+
+
+def test_intersection_two_lines_one_is_point_no_intersection():
+    x_min = np.array([0, 1])
+    x_max = np.array([0, 2])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 0
+
+
+def test_intersection_two_lines_one_is_point_intersection():
+    x_min = np.array([1, 0])
+    x_max = np.array([1, 2])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 2
+    assert pairs[0, 0] == 0
+    assert pairs[1, 0] == 1
+
+
+def test_intersection_three_lines_two_intersect():
+    x_min = np.array([1, 0, 3])
+    x_max = np.array([2, 2, 4])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 2
+    assert pairs[0, 0] == 0
+    assert pairs[1, 0] == 1
+
+
+def test_intersection_three_lines_all_intersect():
+    x_min = np.array([1, 0, 1])
+    x_max = np.array([2, 2, 3])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 6
+    assert check_pairs_contain(pairs, [0, 1])
+    assert check_pairs_contain(pairs, [0, 2])
+    assert check_pairs_contain(pairs, [1, 2])
+
+
+def test_intersection_three_lines_pairs_intersect():
+    x_min = np.array([0, 0, 2])
+    x_max = np.array([1, 3, 3])
+
+    pairs = intersections._identify_overlapping_intervals(x_min, x_max)
+
+    assert pairs.size == 4
+    assert check_pairs_contain(pairs, [0, 1])
+    assert check_pairs_contain(pairs, [1, 2])
+
+
+# ---------- Testing _intersect_pairs ----------
```

Comment:
It should be possible to parametrize this part as well.

Reply:
Agreed, doing it.